api/storage_service.cc

```diff
@@ -780,12 +780,16 @@ void set_storage_service(http_context& ctx, routes& r, sharded<service::storage_
             return make_ready_future<json::json_return_type>(json_void());
         });
     });
-    ss::truncate.set(r, [&ctx](std::unique_ptr<request> req) {
-        //TBD
-        unimplemented();
+    ss::truncate.set(r, [&ctx](std::unique_ptr<request> req) -> future<json::json_return_type> {
         auto keyspace = validate_keyspace(ctx, req->param);
         auto column_family = req->get_query_param("cf");
-        return make_ready_future<json::json_return_type>(json_void());
+        auto ts = db_clock::now();
```

Comment:
Wow, so this was just unimplemented? Why is this patch part of *this* series? Do we have a test for it? I would also open an issue (if we don't have one yet?) and mark this patch "Fixes" it (in case we'll want to backport it, for example).

Reply:
The patch is part of this series just to facilitate testing truncate via the api.
But the test could also issue truncate via CQL.
I'll open an issue about that (and look for other unimplemented API calls)
and we can fix it separately.

Reply:
Good you opened the issue, it lead to a good discussion and I think a conclusion that we don't need this REST API - your test can simply use the CQL "TRUNCATE" command, right?

Reply:
and also 
```python
@contextmanager
def new_test_snapshot(rest_api, keyspaces=[], tables=[], tag=""):
```